eslint.config.mjs

```diff
@@ -37,6 +37,14 @@ invariant(react.configs.flat, 'For typescript');
 invariant(react.configs.flat.recommended, 'For typescript');
 invariant(react.configs.flat['jsx-runtime'], 'For typescript');
 
+// lint rules that need type information need to go here
+export const typeAwareLintRules = {
+  name: 'plugin/typescript-eslint/typed-aware-linting',
+  rules: {
+    '@typescript-eslint/no-unnecessary-type-assertion': 'error',
+  },
+};
+
```

Comment:
I’m exporting the rules that need type information separately here so that we can easily loop over them in the `eslint-pre-commit` config.

`no-unnecessary-type-assertion` is just an example rule to turn on, we can add more here later, but we can’t really use any presets because we want to be able to explicitly turn them off.

Reply:
note: I found an even better way that doesn’t involve exporting the rules. We can just use: 

```
extends: [typescript.configs.disableTypeChecked]
```

which is a built-in way to disable all rules that rely on the type checker